state/execution.go

```diff
@@ -307,19 +307,20 @@ func (blockExec *BlockExecutor) CheckBlock(block *types.Block) error {
 			return fmt.Errorf("invalid transaction, code: %d", tx.Code)
 		}
 	}
-	resultHash := CheckBlockResponseResultHash(resp)
-	if !bytes.Equal(resultHash, block.Header.LastResultsHash.Bytes()) {
+	resultHash := ABCIResponsesResultsHash(&tmstate.ABCIResponses{
+		DeliverBlock: &abcix.ResponseDeliverBlock{Events: resp.Events, DeliverTxs: resp.DeliverTxs}})
+	if !bytes.Equal(resultHash, block.Header.ResultsHash.Bytes()) {
 		blockExec.logger.Error(
-			"resultHash mismatch. ResultHash in ResponseCheckBlock: %X\n ResultHash in block header: %X",
-			resultHash, block.Header.LastResultsHash,
+			"resultHash mismatch.", "ResultHash in ResponseCheckBlock:", resultHash,
```

Comment:
`"resultHash mismatch.", "ResultHash in ResponseCheckBlock:"` can be merged to one string. Also add a space after `:`

Reply:
they are different, the first part is message, the second part is the error's key

Reply:
then colon can be removed in the key